UI/EncyclopediaDetailPanel.cpp

```diff
@@ -939,6 +942,39 @@ void EncyclopediaDetailPanel::HandleLinkDoubleClick(const std::string& link_type
     }
 }
 
+namespace {
+    std::unordered_set<std::string> GetSubDirs(const std::string& dir_name,
+                                               int depth = 1)
+    {
+        std::unordered_set<std::string> retval {};
+        // safety check to pre-empt potential infinite loop
+        const int depth_limit =
+            GetOptionsDB().Get<int>("UI.encyclopedia.search.recursive-limit");
+        if (depth > depth_limit) {
+            WarnLogger() << "Exceeded recursive limit with lookup for pedia"
+                         << " category " << dir_name;
+            return retval;
+        }
+
+        const auto& article_it = GetEncyclopedia().articles.find(dir_name);
+        if (article_it != GetEncyclopedia().articles.end()) {
```

Comment:
can rearrange this to exit if `article_it == ().end` and then de-indent the following code.